pswalker/utils/argutils.py

```diff
@@ -72,3 +73,24 @@ def field_prepend(field, obj):
         field = obj.name
     
     return field
+
+
+def isiterable(obj):
+    """
+    Function that determines if an object is an iterable, not including 
+    str.
+
+    Parameters
+    ----------
+    obj : object
+        Object to test if it is an iterable.
+
+    Returns
+    -------
+    bool : bool
+        True if the obj is an iterable, False if not.
+    """
+    if isinstance(obj, str):
+        return False
+    else:
+        return isinstance(obj, Iterable)
```

Comment:
We definitely could, we would just need to come up with sensible way to handle when the ioc runs assuming we only wanting it running while we do the alignment.